File: RedSheet.DbEntities/Models/EventAgendaTiming.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
    [Table("EventAgendaTimings", Schema = "dbo")]
    [Serializable]
    public partial class EventAgendaTiming
    {

        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public int EventAgendaTimingId { get; set; }

        [Required]
        [MaxLength(200)]
        public string PayOff { get; set; }

        [Required]
        [MaxLength(200)]
        public string Process { get; set; }

        [Required]
        [MaxLength(200)]
        public string Purpose { get; set; }

        [Range(1, int.MaxValue)]
        public int SortOrder { get; set; }

        [Required]
        public TimeSpan Time { get; set; }

        [Required]
        [MaxLength(200)]
        public string Topic { get; set; }

        [Required]
        [MaxLength(200)]
        public string Trigger { get; set; }

        [RelationshipTableAttribue("ApplicationObjects", "dbo")]
        public NegotiationPhase NegotiationPhaseId { get; set; }

        [RelationshipTableAttribue("ProjectEventTimelines", "dbo")]
        //Relationships
        public int ProjectEventTimelineId { get; set; }

        [ForeignKey("ProjectEventTimelineId")]
        public ProjectEventTimeline ProjectEventTimeline { get; set; }

        [RelationshipTableAttribue("Tactics", "dbo")]
        //Relationships
        public int TacticsId { get; set; }

        [ForeignKey("TacticsId")]
        public Tactic Tactic { get; set; }

        public EventAgendaTiming()
        {
        }

    }
}

File: redsheet-portal/src/app/database-models/event-agenda-timing.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectEventTimeline, Tactic, vEventAgendaTimingRecord } from './'
export class EventAgendaTiming {
    constructor(eventAgendaTiming?: EventAgendaTiming | vEventAgendaTimingRecord) {
        let properties = ["eventAgendaTimingId", "payOff", "process", "purpose", "sortOrder", "time", "topic", "trigger", "negotiationPhaseId", "projectEventTimelineId", "tacticsId", "previousEventAgendaTimingId", "previousEventAgendaTimingSortOrder", "previousEventAgendaTimingTime",];
        for (let property of properties)
            if (eventAgendaTiming && eventAgendaTiming[property])
                this[property] = eventAgendaTiming[property];
    }

    eventAgendaTimingId: number = 0;

    @required()
    @maxLength(200)
    payOff: string = undefined;

    @required()
    @maxLength(200)
    process: string = undefined;

    @required()
    @maxLength(200)
    purpose: string = undefined;

    @range(1, 2147483647)
    sortOrder: number = undefined;

    @required()
    time: Date = undefined;

    @required()
    @maxLength(200)
    topic: string = undefined;

    @required()
    @maxLength(200)
    trigger: string = undefined;

    @range(0, 2147483647)
    negotiationPhaseId: number = undefined;

    @range(0, 2147483647)
    projectEventTimelineId: number = undefined;
    projectEventTimeline: ProjectEventTimeline;

    @range(0, 2147483647)
    tacticsId: number = undefined;
    tactic: Tactic;

    previousEventAgendaTimingId: number = undefined;
    previousEventAgendaTimingSortOrder: number = undefined;
    previousEventAgendaTimingTime: string = undefined;

}
